### PR TITLE
cleanup .travis.yml and associated files

### PR DESCRIPTION
**Pull request by jonlawlor**
As explained in https://groups.google.com/d/msg/gonum-dev/GKJjAtDxR9I/n0FD7WvvN5AJ

Removed notifications and go tip, simplified code, added some comments, put all travis related code (except for .travis.yml) into .travis/

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,33 +1,20 @@
...
-
-env:
-  global:
-  - secure: "bbvnev0lw5pf9wY/Fv9CgDCynS0q8wvAiRXC/wPwaTR/X7bWwuQF7YmBhOlS9H/x1IxHW1HyqJMjOipIF2/MgPoTB4xVkk1jLHigd6MCt91o8aZvkaW4ITDKhP29CZNCyBKvPoEs/yLGEfaGYop3ivfJPWnDnMAd04m7+Yr1+3I="
-  
 
+# After go 1.3, go tools moved from code.google.com to golang.org
 before_install:
- - if ! go get code.google.com/p/go.tools/cmd/cover; then go get golang.org/x/tools/cmd/cover; fi
+ - if ! go get golang.org/x/tools/cmd/cover; then go get code.google.com/p/go.tools/cmd/cover; fi
```
Comment (kortschak) [inline .travis.yml]: Is it worth keeping code.google.com/p/go.tools/...?

**Comment by kortschak** [comment]
LGTM but consider comment.

**Comment by jonlawlor** [comment]
You're right, it works without code.google.com.

**Comment by kortschak** [comment]
I don't think you can actually pull from there anymore.

LGTM

**Comment by kortschak** [comment]
Can this be rolled out to the other repos? It looks like no repo-specific changes are needed. I'm happy to do that if you want, otherwise I'll wait for you.

**Comment by jonlawlor** [comment]
That's the plan, although there is some repo specific stuff for blas, lapack(?), and matrix.  I wanted to do this repo first because it seems mostly inactive.

I'm probably going to have time to finish things up tonight, failing that then on the next two days.  If you want to go ahead with the same changes I'm fine with that.  

You'll have to get the coveralls repo key by going to coveralls.io/r/gonum/{repo} (in this case, https://coveralls.io/r/gonum/diff).  If you are logged in, which you can do via your github account, then you'll see the repo token on the right.  Add secure environment variable to the travis build by going to the travis page for the repo (in this case, https://travis-ci.org/gonum/diff) and then clicking on settings->settings, pressing "ENVIRONMENT VARIABLES", selecting "Add a new variable", and then give it a name of COVERALLS_TOKEN and the value should be the repo token from coveralls.

For matrix, I figure a folder structure more like
`.travis/{arch}/{blas provider}` would be sensible.  Travis still says it isn't accepting multi-arch builds, so arch will always be linux, and {blas provider} would be either gonum or OpenBLAS.  The shell scripts are going to take a bit more reworking.

**Comment by kortschak** [comment]
I suspect that by the time I get to it you may already have been there - long weekend and child (with a newly acquired fascination for lego!).

**Pull request merged**